src/Analyzers/MSTest.GlobalConfigsGenerator/MSTestAnalysisMode.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+internal enum MSTestAnalysisMode
+{
+    /// <summary>
+    /// Disables all analyzers. User will need to opt-in to all analyzers.
+    /// </summary>
+    None,
+
+    /// <summary>
+    /// Follow the current behavior.
+    /// Rules that are enabled by default as warnings remain enabled as warnings. Anything else isn't enabled by default.
```

Comment:
Maybe we can simplify by saying we respect definition of the analyzer (enabled by default + severity).